File: Java/PivotIndex.java
class Solution {
    public int pivotIndex(int[] nums) {
        int pivotIndex = -1;
        int[] sumArr = new int[nums.length+1];
        sumArr[0] = 0;
        for(int i=0; i< nums.length; i++){
            sumArr[i+1] = sumArr[i]+ nums[i]; 
        }
        sumArr[nums.length] = sumArr[nums.length-1] + nums[nums.length-1];
        int rightIndex = sumArr.length-1;
        for(int i = 0;i<sumArr.length-1;i++){
            if(sumArr[i] == (sumArr[rightIndex]-sumArr[i+1])){
               return i;
            }
        }
        return pivotIndex;
    }
}

File: JavaScript/Array/PivotIndex.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function(nums) {
    // time compexity - O(n)
    // space complexity - O(n)
    let n = nums.length;
    let dpArr = new Array(n).fill(0);
    
    for(let i=1; i<=n; i++){
        dpArr[i] = dpArr[i-1]+ nums[i-1];
    }
    for(let i=0; i<=n; i++){
        if(dpArr[i] == (dpArr[n] - dpArr[i+1])){
            return i;
        }
    }
    return -1;
};
